packages/identity/src/idp-identity.ts

```diff
@@ -0,0 +1,339 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+export * from "./cdk-internals";
+
+import {
+  IdentityPool,
+  IdentityPoolProps,
+  UserPoolAuthenticationProvider,
+} from "@aws-cdk/aws-cognito-identitypool-alpha";
+import { Duration, RemovalPolicy, Annotations } from "aws-cdk-lib";
+import {
+  UserPoolIdentityProviderAmazon,
+  UserPoolIdentityProviderApple,
+  UserPoolIdentityProviderFacebook,
+  UserPoolIdentityProviderGoogle,
+  UserPoolIdentityProviderOidc,
+  UserPoolIdentityProviderSaml,
+  PasswordPolicy,
+  UserPool,
+  UserPoolClient,
+  AccountRecovery,
+  CfnUserPool,
+  Mfa,
+  // UserPoolIdentityProvider,
+  IUserPoolIdentityProvider,
+  UserPoolClientIdentityProvider,
+  UserPoolProps,
+  CognitoDomainOptions,
+  CustomDomainOptions,
+  UserPoolDomain,
+  IUserPool,
+  UserPoolClientOptions,
+} from "aws-cdk-lib/aws-cognito";
+import { Construct } from "constructs";
+import {
+  UserPoolIdentityProviderAmazonProps,
+  UserPoolIdentityProviderAppleProps,
+  UserPoolIdentityProviderFacebookProps,
+  UserPoolIdentityProviderGoogleProps,
+  UserPoolIdentityProviderOidcProps,
+  UserPoolIdentityProviderSamlProps,
+} from "./cdk-internals";
+import { IdentityProviderName } from "./cdk-internals/identityProviders";
+
+export const defaultPasswordPolicy: PasswordPolicy = {
+  minLength: 8,
+  requireLowercase: true,
+  requireUppercase: true,
+  requireDigits: true,
+  requireSymbols: true,
+  tempPasswordValidity: Duration.days(3),
+};
+
+export interface IdentityProviderProps {
+  readonly [IdentityProviderName.AMAZON]?: UserPoolIdentityProviderAmazonProps;
+  readonly [IdentityProviderName.APPLE]?: UserPoolIdentityProviderAppleProps;
+  readonly [IdentityProviderName.FACEBOOK]?: UserPoolIdentityProviderFacebookProps;
+  readonly [IdentityProviderName.GOOGLE]?: UserPoolIdentityProviderGoogleProps;
+  readonly [IdentityProviderName.OIDC]?: UserPoolIdentityProviderOidcProps;
+  readonly [IdentityProviderName.SAML]?: UserPoolIdentityProviderSamlProps;
+}
+
+/**
+ * Properties which configures the IDP Identity construct.
+ */
+export interface IdpIdentityProps {
+  /**
+   * User provided Cognito UserPool.
+   *
+   * @default - a userpool will be created.
+   */
+  readonly userPool?: IUserPool;
+
+  /**
+   * Props for the UserPool construct
+   */
+  readonly userPoolProps?: UserPoolProps;
```

Comment:
nit/TILI: rename to defaultUserPoolProps.

Reply:
I wonder if it would be better to simplify and to not have this at all - like the `UserIdentity` construct we could allow customisation by passing the user pool instance?

Reply:
there are instances where the user may only want to change one prop - should we force them to "manager" their own UserPool as a tax ?

Reply:
They must pay their penance! 😂 Maybe not, but might be good to have consistent behaviour between the different identity constructs PDK offers :)